WikiWatershed/model-my-watershed
# Replace SJS with Akka HTTP server

## Overview

This PR removes SJS and replaces it with the new Akka HTTP jar here https://github.com/WikiWatershed/mmw-geoprocessing/releases/tag/3.0.0-alpha

It also adds an `akka-http` upstart job to the Worker VM

Connects #2101 
Connects #2103 

## Testing Instructions
- get branch, then `vagrant reload worker --provision`
- verify that it provisions correctly
- once that's done, make a get request to `:8090/ping` and a post request to `:8090/run` from host, verifying that each returns a response with headers including the line:

```
Server: akka-http/10.0.9
```

- ssh into the Worker VM, then try the following commands and verify that they all work correctly:

```
sudo restart geoprocessing
sudo stop geoprocessing
sudo start geoprocessing
```

- `sudo less /var/log/upstart/geoprocessing.log` and verify that you see the log. Possibly worth `tail -f` to watch it while starting, stopping, restarting, and using the service, too, to verify that the log's getting updated.
